crud-rest-service/src/users/users.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  Query,
+  Res,
+  HttpStatus,
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common';
+import { UsersService } from './users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { PaginationQueryDto } from 'src/common/dto/pagination-query.dto';
+import { LoginSubstringUserDto } from './dto/getLoginSubstring-user.dto';
+
+@Controller('v1/users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Post()
+  create(@Body() createUserDto: CreateUserDto, @Res() response) {
+    const user = this.usersService.findByLogin(createUserDto.login);
+    if (user) {
+      throw new BadRequestException(
+        `User ${createUserDto.login} already exists in the database`,
+      );
+    }
+    const newUser = this.usersService.create(createUserDto);
+    return response.status(HttpStatus.OK).send(newUser);
+  }
+
+  @Get('/limitusers')
+  getAutoSuggestUsers(
+    @Body() loginSubstringUserDto: LoginSubstringUserDto,
+    @Res() response,
+  ) {
+    const users = this.usersService.getAutoSuggestUsers(loginSubstringUserDto);
+    return response.status(HttpStatus.OK).send(users);
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string, @Res() response) {
+    const user = this.usersService.findOne(id);
+    if (!user) {
+      throw new NotFoundException("User does'n exist");
+    }
+    return response.status(HttpStatus.OK).send(user);
+  }
+
+  @Get()
+  findAll(@Query() paginationQuery: PaginationQueryDto) {
+    const { limit, offset } = paginationQuery;
+    return this.usersService.findAll(limit, offset);
+  }
+
+  @Patch(':id')
```

Comment:
Actually I think, it's not a problem, but may be we should use PUT method instead of PATCH?)

Reply:
I also think that these 2 methods are almost equivalent, but I can replace.